### PR TITLE
🖲 811 implement rpc triggering method for test creation

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the create test flow for the RPC triggering method. The Backend is not wired up yet as the service is not ready to support this new triggering method.

## Changes

- Adds RPC triggering method.

## Fixes

- https://github.com/kubeshop/tracetest/issues/811 (Partially) 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/e60da3fea1964bac8f75ecfb48a15357

### CODE REVIEW

File: web/src/components/CreateTestPlugins/Rest/steps/RequestDetails/RequestDetailsAuthInput/TypeInput.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const methodNamingMap: Record<string, string> = {
 };
 
 interface IProps {
-  form: FormInstance<IRequestDetailsValues>;
```
Comment (xoscar) [inline web/src/components/CreateTestPlugins/Rest/steps/RequestDetails/RequestDetailsAuthInput/TypeInput.tsx:13]: Moving this to a more generic type

File: web/src/components/CreateTestPlugins/Rest/steps/RequestDetails/RequestDetailsUrlInput.tsx [hunk 2]
```diff
@@ -1,30 +1,35 @@
...
-                return (
-                  <Select.Option data-cy={`method-select-option-${method}`} key={method} value={method}>
-                    {method}
-                  </Select.Option>
-                );
-              })}
-            </Select>
-          </Form.Item>
-        </div>
+        {showMethodSelector && (
```
Comment (xoscar) [inline web/src/components/CreateTestPlugins/Rest/steps/RequestDetails/RequestDetailsUrlInput.tsx:13]: New prop to check if it needs to show the method input

File: web/src/components/CreateTestPlugins/Rpc/steps/RequestDetails/RequestDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    (values: IRequestDetailsValues) => {
+      console.log('@@values', values);
+      onNext();
+    },
+    [onNext]
+  );
+
+  const protoFile = Form.useWatch('protoFile', form);
+
+  const getMethodList = useCallback(async () => {
```
Comment (xoscar) [inline web/src/components/CreateTestPlugins/Rpc/steps/RequestDetails/RequestDetails.tsx:39]: Updates the valid method list based from the Protobuf file

File: web/src/services/Rpc.service.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import {parse, NamespaceBase, Service} from 'protobufjs';
+
+const RpcService = () => ({
```
Comment (xoscar) [inline web/src/services/Rpc.service.ts:3]: Reads the protobuf file and extracts the available methods.

**Review by jorgeepc — APPROVED** [review]
Great work. Clean 🔥

**Review by cescoferraro — APPROVED** [review]
LGTM

**Pull request merged**